### DISCUSSION TITLE
Sum search results or values in set date range?

### DISCUSSION THREAD

**Discussion by thehijacker** [firefly-iii/firefly-iii#9174] [Firefly III] [answered]
I have persuaded wife to manage her transactions using Firefly III. Done the import from bank from start of year until now and made all the automation rules. She was still not 100 % sure if she would use this until I showned her how much she spents in shops :).

Now she wants to check per date range and with search results how much she spent, but search results do not get summed. The same goes with results from Classification->Categories

We went to shops on Friday. She added all the transaction with (Shop Name) and they get put in same Shop category. How would she get sum of Category named Shop only for that one day (or a week, or whatever date range she has set)? All I see is sum per whole month.

![image](https://github.com/user-attachments/assets/ea72d835-2eaa-4847-907a-d2d756034736)

And same for search result. For example if she searches for specific shop name she gets all transactions in results but she is missing sum of all values:

![image](https://github.com/user-attachments/assets/f87ce365-457f-49fb-b951-8cf6698182f8)

Thank you for the tips. Hopefully this is already implemented inside, I just do not know how to find this. It would increase the WAF. :)

**Comment by JC5** [comment, marked as answer]
Hey, good question.

Check out the reports page :). You probably want the category report.

**Reply by thehijacker** [reply]
Just checked this. Category report works for categories and can use the date range. I will show her this.

Is there a chance to get sum of transactions from search results?

**Reply by JC5** [reply]
Yea, there is a ticket for it already 😃

**Reply by RatWolf** [reply]
I also wish for that, since I want a sum on all reconciliations within a time-range or from all accounts containing a keyword (like work1 work2 and so on).
I mean, new filter for reports or the ability to create new reports also would be great.

**Comment by niklas152** [comment]
With the obvious disclaimer to NEVER run code in your browser you got from somewhere random in the internet, especially if you don't fully, 100% understand and trust it and its author:

Here is a javascript function that can be used to sum up the values in a search. It can only work on the ones visible, thus for searches that return so many results that it paginates, it won't work.

```JavaScript
// get the div of the search result, (as of now) its second child is the results table, get the rows of that
let rows = [...document.getElementsByClassName('search_results')[0].children[1].rows];
let sum = 0.0;
rows.forEach(function(row, row_idx, rows) {
	let cells = [...row.cells];
	// result rows (currently) always have 9 entries, and we don't care about the table header
	if (cells.length != 9 || cells[2].nodeName === 'TH') {
		return;
	}
	// transform the text to a number
	let saldoString = String(cells[2].innerText);
	let saldoValue = Number(saldoString.split(/(\s+)/)[0].trim().replace('.','').replace(',', '.'));
	// sum it up
	sum += saldoValue;
}); 
// print it out
alert(`Sum in search: ${sum}€`);
```

You can also create a bookmark with the function wrapped like in the following snippet as the URL of the bookmark to have it easily accesible in your browser:

```JavaScript
javascript:(function() { let rows = [...document.getElementsByClassName('search_results')[0].children[1].rows]; let sum = 0.0;	rows.forEach(function(row, row_idx, rows) { let cells = [...row.cells]; if (cells.length != 9 || cells[2].nodeName === 'TH') { return; } let saldoString = String(cells[2].innerText); let saldoValue = Number(saldoString.split(/(\s+)/)[0].trim().replace('.','').replace(',', '.')); sum += saldoValue; }); alert(`Sum in search: ${sum}€`); }());
```

Again, running code in your browser from some stranger on the internet is almost never a good idea. It also doesn't work with pagination in search and it could break at any moment as it is not using any official API but just scrapes the info from the current state of the HTML, which could potentially (and rightfully so) change with any update. It's also in no way polished. It will also only work for the search, not the categories page (although something similar could be build for that page, too).
But, with all that, it might be a temporary workaround.

**Reply by RatWolf** [reply]
Thanks, I edited it a bit to get it shown under the Table and now the only thing what I'm missing is, to add the original URL to the bookmark, so it calls the page and after that inserts it. But that won't work right now... But so far, it's looking really good!
![image](https://github.com/user-attachments/assets/1341af49-1015-463d-8c13-1e4d91118dba)
On my own, I wouldn't know where to start and how everything works, but when I've got such good code I could modify it, so thank you so much for the help so far!
